backend/small_eod/api/urls.py

```diff
@@ -0,0 +1,18 @@
+from django.urls import include, path
+from rest_framework import routers
+
+from api.views import UserViewSet, AddressDataViewSet, InstitutionViewSet, CaseViewSet
+from dictionary.views import DictionaryViewSet
+router = routers.DefaultRouter()
+router.register('users', UserViewSet)
+router.register('address', AddressDataViewSet)
+router.register('institution', InstitutionViewSet)
+router.register('case', CaseViewSet)
+router.register('dictionary', DictionaryViewSet)
+
+# Wire up our API using automatic URL routing.
+# Additionally, we include login URLs for the browsable API.
+urlpatterns = [
+    path('', include(router.urls)),
+    path('api-auth/', include('rest_framework.urls', namespace='rest_framework'))
+]
```

Comment:
Proponuje przyjąć strukturę plików:

* back-end
  * small_eod
     * dictionary
     * cases
     * letters
     * events
  * config
     * urls.py
     * settings
        * local.py
        * base.py
        * common.py

Wówczas spinanie routingu będzie w ```config/urls.py``` zamiast w arbitralnie wybranej aplikacji, która zawiera różne rzeczy.

Zobacz https://github.com/watchdogpolska/poradnia/blob/master/config/urls.py i https://github.com/pydanny/cookiecutter-django/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/config/urls.py (pydanny to dobra inspiracja, autor ma doświadczenie, możesz spojrzeć także na jego książki: https://www.pydanny.com/pages/books/ ).

Reply:
Ten cookie cutter wyglada swietnie. Przygladne sie temu blizej. 

Jezeli chodzi o urle - to w obydwu projektach jest struktura oparta o zwykle class based views. Jezeli my dla API chcemy uzywac viewsetow, to z tego co widzialem, najlatwiej je spinac z routerem. I stad moje pytanie: Czy definiuje sie router dla kazdej aplikacji, czy wystarczy jeden globalny w config/urls gdzie beda wszystkie veiwsety importowane?

Struktura zmieniona - sciezki dostosowane, aplikacje zmienione na liczby mnogie (case -> cases)

Reply:
Nie mam doświadczenia z routerami, aby dać Ci odpowiedź. Zobacz na projekt netbox w jaki sposób oni do tego podeszli.

Reply:
collections zmienione na case_collections zeby nie kolidowalo z wbudowana biblioteka

Reply:
A może zastąpimy to zupełnie innym słowem, które określi zbiór, kolekcje, zestaw, skoro już teraz wywołuje problem?

Reply:
Tak w ogóle to jaki problem występuje z collections? W przypadku Python masz namespaces, więc skoro ```collections``` nie jest zarezerwowanym słowem kluczowym to nei jest problemem. https://code.tutsplus.com/tutorials/what-are-python-namespaces-and-why-are-they-needed--cms-28598 
Dzięki temu ```Case``` występuje w Django w ```django.db.models.Case``` i nie koliduje.

Reply:
Przed zmiana struktury projektu, jak dodalem 'collections' w INSTALLED_APPS, to mi wywalalo bledy ktore sugerowaly, ze django probuje importowac ze standardowej biblioteki pythona. Teraz jest 'small_eod.case_collections' i nie ma problemu, tak samo prawdopodobnie nie bedzie problemu z 'small_eod.collections' - daj znac jak wolisz, zeby bylo. Co do innego slowa, 'aggregations' - nic innego mi do glowy nie przychodzi